### PR TITLE
Added rpc call to fetch finality proof and submit it to pallet-feeds

### PR DESCRIPTION
**Pull request by isSerge**

### CODE REVIEW

File: backend/src/relay.ts [hunk 1]
```diff
@@ -172,7 +176,20 @@ export default class Relay {
...
       );
 
-      const extraBytes = block.byteLength + metadata.byteLength;
+      let extraBytes;
+      let proof;
+
+      // get proof for relay chain block and add it to bytes
+      if (isRelayChain) {
+        proof = hexToUint8Array((await pRetry(
+          () => this.sourceApi.rpc.grandpa.proveFinality(11445772),
```
Comment (nazar-pc) [inline backend/src/relay.ts:185]: Is this supposed to be hardcoded number?

File: backend/src/test/relay/common.ts [hunk 2]
```diff
@@ -48,7 +48,7 @@ export const setup = async () => {
   await cryptoWaitReady();
 
   const signer = new PoolSigner(
-    registry,
+    registry as unknown as Registry,
```
Comment (nazar-pc) [inline backend/src/test/relay/common.ts:51]: Why is this necessary?

**Review by nazar-pc — COMMENTED** [review]
Do you want to merge it right away? I'm suspecting it may not be compatible with current node version due to extra call argument.

**Comment by isSerge** [comment]
@nazar-pc no, I won't merge it right away. I will merge it after I merge changes for pallet-feeds

File: backend/src/test/relay/common.ts (hunk 2 above)
Comment (isSerge) [reply backend/src/test/relay/common.ts:51]: yes, due to changes in the Polkadot.js API

**Comment by nazar-pc** [comment]
I converted this to draft for now

File: backend/src/test/relay/common.ts (hunk 2 above)
Comment (nazar-pc) [reply backend/src/test/relay/common.ts:51]: Hm, what exactly has changed there? I would imagine to registry be just registry.

Comment (isSerge) [reply backend/src/test/relay/common.ts:51]: After update: `Argument of type 'TypeRegistry' is not assignable to parameter of type 'Registry'`. This did not affect any application code, only this line in tests

Comment (nazar-pc) [reply backend/src/test/relay/common.ts:51]: Maybe you need to do `new Registry` instead of `new TypeRegistry` now?

Comment (isSerge) [reply backend/src/test/relay/common.ts:51]: No, I tried that. I can investigate that

File: backend/src/poolSigner.ts [hunk 3]
```diff
@@ -59,7 +60,7 @@ export class PoolSigner extends SignerWithAddress {
     private readonly workers: SigningWorker[] = [];
 
     public constructor(
-        private readonly registry: Registry,
+        private readonly registry: TypeRegistry | Registry,
```
Comment (nazar-pc) [inline backend/src/poolSigner.ts:63]: Is `Registry` still necessary here?

Comment (isSerge) [reply backend/src/poolSigner.ts:63]: Yes, we use both in diff places

File: backend/src/relay.ts [hunk 4]
```diff
@@ -185,12 +185,12 @@ export default class Relay {
           createRetryOptions(error => this.logger.error(error, `get block justifications for #${nextBlockToProcess} retry error:`)),
         );
         const bytes = proof.toU8a(true);
-        const decodedProof = decodeProof(bytes);
+        const justification = bytes.slice(36);
```
Comment (nazar-pc) [inline backend/src/relay.ts:188]: You definitely need to add a comment about this magic number, better yet to make it a named constant and explain what it means and why we use it.

Comment (isSerge) [reply backend/src/relay.ts:188]: sure, I want to refactor this method - it's getting messy

File: backend/src/relay.ts [hunk 5]
```diff
@@ -224,6 +204,30 @@ export default class Relay {
...
+    // adding justifications from finality proof if there is none
+    if (!rawBlock.justifications) {
+      const { number } = rawBlock.block.header;
+      const proof = await pRetry(
+        () => this.sourceApi.rpc.grandpa.proveFinality(number),
+        createRetryOptions(error => this.logger.error(error, `get block justifications for #${number} retry error:`)),
+      );
+      const proofBytes = proof.toU8a(true);
+      // finality proof returns encoded block hash, justification and unknown headers
+      // only care about justification, removing leading extra bytes: 32 for block hash and 4 for SCALE prefix(?)
```
Comment (nazar-pc) [inline backend/src/relay.ts:217]: Number of bytes for scale codec should ideally be determined dynamically and not assumed that it is 4 bytes. You can just decode it as a vector and read it properly that way.

File: backend/src/relay.ts [hunk 6]
```diff
@@ -224,6 +204,30 @@ export default class Relay {
...
+      const justification = proofBytes.slice(36);
+
+      rawBlock.justifications = [
+        [
+          [70, 82, 78, 75], // FRNK
```
Comment (nazar-pc) [inline backend/src/relay.ts:222]: What is `FRNK`?

Comment (isSerge) [reply backend/src/relay.ts:222]: It's engine ID, there are two of them in consensus logs: BABE and FRNK. I don't know the story behind the name, but this is the one we need for GRANDPA verification

Comment (nazar-pc) [reply backend/src/relay.ts:209]: Named constant like this would be more readable:
```typescript
const GRANDPA_ENGINE_ID = [70, 82, 78, 75]; // FRNK
```
See https://github.com/paritytech/substrate/blob/ded44948e2d5a398abcb4e342b0513cb690961bb/primitives/finality-grandpa/src/lib.rs#L57-L58
